tools/VscPreviewExtension/src/childProcessHost.ts

```diff
@@ -74,24 +61,59 @@ export class ChildProcessHost {
         let docContent = doc.getText();
         let fileName = doc.fileName;
         let rootPath = workspace.rootPath;
-        let filePath;
+        let relativePath;
         if (!rootPath || !fileName.includes(rootPath)) {
-            let indexOfFilename = fileName.lastIndexOf("\\");
-            rootPath = fileName.substr(0, indexOfFilename);
-            filePath = fileName.substring(indexOfFilename + 1);
+            let indexOfFileName = fileName.lastIndexOf("\\");
+            rootPath = fileName.substr(0, indexOfFileName);
+            relativePath = fileName.substring(indexOfFileName + 1);
         } else {
             let rootPathLength = rootPath.length;
-            filePath = fileName.substr(rootPathLength + 1, fileName.length - rootPathLength);
+            relativePath = fileName.substr(rootPathLength + 1, fileName.length - rootPathLength);
         }
-        if (doc.languageId === ConstVariable.languageId) {
-            let numOfRow = doc.lineCount;
-            this.writeToStdin(rootPath, filePath, numOfRow, docContent);
+        if (doc.languageId === "markdown") {
+            this.sendHttpRequestCoreAsync(rootPath, relativePath, docContent);
         }
     }
 
-    protected writeToStdin(rootPath: string, filePath: string, numOfRow: number, docContent: string) { }
+    protected async sendHttpRequestCoreAsync(rootPath: string, relativePath: string, docContent: string) {
+        window.showErrorMessage(`[Extension Error]: Not supported`);
+    }
+
+    protected newHttpServerAndStartPreview(activeTextEditor) {
+        if (this._isChildProcessStarting)
+            return;
+        this._isChildProcessStarting = true;
+        window.showInformationMessage("Environment initializing, please wait several seconds!");
+        let that = this;
+        let http = require("http");
+        let server = http.createServer();
+        server.listen(0);
+        server.on('listening', function () {
+            var port = server.address().port;
+            server.close();
+            that.newHttpServerAndStartPreviewCore(activeTextEditor, port);
```

Comment:
>newHttpServerAndStartPreviewCore [](start = 17, length = 32)

after port is get from server, should close the server and exist the function immediatedly, then to call newHttpServerAndStartPreviewCore?